core/shared/src/main/scala/zio/prelude/AssociativeFlatten.scala

```diff
@@ -240,4 +240,62 @@ trait AssociativeFlattenSyntax {
     def flatMap[B](f: A => F[B])(implicit flatten: AssociativeFlatten[F], covariant: Covariant[F]): F[B] =
       flatten.flatten(covariant.map(f)(fa))
   }
+
+  /**
+   * Provides infix syntax for filtering covariant types.
+   */
+  implicit class AssociativeFlattenCovariantIdentityBothIdentityEitherOps[F[+_], A](fa: F[A]) {
+
+    /**
+     * Collects values of `A` for which the partial function `pf` is defined.
+     */
+    def collect[B](pf: PartialFunction[A, B])(implicit
+      flatten: AssociativeFlatten[F],
+      covariant: Covariant[F],
+      identityBoth: IdentityBoth[F],
+      identityEither: IdentityEither[F]
+    ): F[B] =
+      fa.flatMap(a => pf.lift(a).fold[F[B]](identityEither.none)(_.succeed))
+
+    /**
+     * Collects values of `A` for which the partial function `pf` is defined.
+     */
+    def collectM[B](pf: PartialFunction[A, F[B]])(implicit
+      flatten: AssociativeFlatten[F],
+      covariant: Covariant[F],
+      identityBoth: IdentityBoth[F],
+      identityEither: IdentityEither[F]
+    ): F[B] =
+      fa.flatMap(a => pf.lift(a).getOrElse(identityEither.none))
+
+    /**
+     * Filters an `F[A]` value with a predicate `f`.
+     */
+    def filter[B](f: A => Boolean)(implicit
+      flatten: AssociativeFlatten[F],
+      covariant: Covariant[F],
+      identityBoth: IdentityBoth[F],
```

Comment:
Do we really need `IdentityBoth` here?

Reply:
Yes we can relax this constraint!

Reply:
I still see it here :smile: 
Btw, can you also double check other methods, that they don't contain unnecessary implicits?

Reply:
We need it for `succeed.